descartes_planner/src/ladder_graph_dijkstras.cpp

```diff
@@ -0,0 +1,125 @@
+#include "descartes_planner/ladder_graph_dijkstras.h"
```

Comment:
This code is a combination of the boost algorithm referenced in the intro (never looked at the source) and a blog I found talking about the speed of boost heap implementations: http://stegua.github.io/blog/2012/09/19/dijkstra/. Some of this code was taken from there (specifically the idea of the `ValueKey`, the typedefs, and the `handle` code). I need to give credit to both sources. There is no license included with the blog. I'll contact the author.